app/models/invitation.rb

```diff
@@ -67,15 +67,22 @@ def send_and_set_date(resend: false)
       UserMailer.invitation(self.id).deliver_now
     end
 
-    date_column = resend ? :resent_at : :sent_at
     # Skip callbacks within after_save by using update_column to avoid a callback loop
-    self.update_column(date_column, Time.current)
+    if resend
+      attrs = { resent_at: Time.current }
+      # This applies to old invites when AO3-6094 wasn't fixed.
+      attrs[:sent_at] = self.updated_at if self.sent_at.nil?
+      self.update_columns(attrs)
```

Comment:
I'm checking with PAC and Support chairs to make sure it's okay if we add a sent_at date. If it is, we probably want a test to make sure it's set correctly.

Also, it just occurred to me -- is this going to be the updated_at date that's on the invitation before we resend it? Or the updated_at date that's on the invitation _after_ we resend it (which would basically match the resent_at date, I believe)? I feel like the previous commit did the latter and this one does the former, but I'm not 100% sure.

Reply:
Using `update_column` doesn't update `updated_at` so I don't think so.

Reply:
Oh, you're right, I didn't notice it was update_column! Now I need to totally revisit what I told everyone. 🤦 Thanks for pointing that out.